lib/rubocop/cop/discourse/plugins/no_monkey_patching.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Discourse
+      module Plugins
+        # Don’t monkey-patch classes directly in `plugin.rb`. Instead, define
+        # additional methods in a dedicated mixin (an ActiveSupport concern for
+        # example) and use `prepend` (this allows calling `super` from the mixin).
+        #
+        # If you’re just adding new methods to an existing serializer, then use
+        # `add_to_serializer` instead.
+        #
+        # @example generic monkey-patching
+        #   # bad
+        #   ::Topic.class_eval do
+        #     has_many :new_items
+        #
+        #     def new_method
+        #     end
+        #   end
+        #
+        #   # good
+        #   module MyPlugin::TopicExtension
+        #     extend ActiveSupport::Concern
+        #
+        #     prepended do
+        #       has_many :new_items
+        #     end
+        #
+        #     def new_method
+        #     end
+        #   end
+        #
+        #   ::Topic.prepend(MyPlugin::TopicExtension)
```

Comment:
This example is not zeitwerk-friendly. I think it should be something like

```
reloadable_patch { Topic.prepend(MyPlugin::TopicExtension) }
```
?

Reply:
It was just an example to show the usage of `prepend`, but yes, I guess it won’t hurt to add what you suggest :slightly_smiling_face: